src/main/java/cientistavuador/ciencraftreal/event/Handler.java

```diff
@@ -0,0 +1,99 @@
+package cientistavuador.ciencraftreal.event;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class Handler<T extends Event> {
+
+    protected static final List<Handler<?>> HANDLER_LIST = new ArrayList<>();
+    protected final List<Listener> listeners = new ArrayList<>();
+
+    /**
+    * Construct a {@link Handler}.
+    */
+
+    public Handler() {
+        HANDLER_LIST.add(this);
+    }
+
+    /**
+    * @return ...
+    */
+
+    public List<Listener> getListeners() {
+        return this.listeners;
+    }
+
+    /**
+    * ...
+    *
+    * @param listener ...
+    */
+
+    public void unregister(Listener listener) {
+        this.listeners.remove(listener);
+    }
+
+    /**
+    * ...
+    *
+    * @param listeners ...
+    */
+
+    public void unregisterAll(Collection<Listener> listeners) {
+        this.listeners.removeAll(listeners);
+    }
+
+    /**
+    * ...
+    *
+    * @param listener ...
+    */
+
+    public void register(Listener listener) {
+        this.listeners.add(listener);
+    }
+
+    /**
+    * ...
+    *
+    * @param listeners ...
+    */
+
+    public void registerAll(Collection<Listener> listeners) {
+        this.listeners.addAll(listeners);
+    }
+
+    /**
+    * ...
+    *
+    * @param event ...
+    */
+
+    public void sendEvent(Event event) {
+        for(Listener listener : listeners) {
+            for(Method method : listener.getClass().getMethods()) {
+                if(method.isAnnotationPresent(Listener.Handler.class)) {
+                    if(method.getParameterCount() == 1 && method.getParameterTypes()[0].isAssignableFrom(event.getClass())) {
+                        try {
+                            method.invoke(listener, event);
+                        } catch (InvocationTargetException | IllegalAccessException e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Vê se funciona isso ai meu bom, mó bonito

```suggestion
        listeners.stream()
    .flatMap(listener -> Arrays.stream(listener.getClass().getMethods()))
    .filter(method -> method.isAnnotationPresent(Listener.Handler.class))
    .filter(method -> method.getParameterCount() == 1 && method.getParameterTypes()[0].isAssignableFrom(event.getClass()))
    .forEach(method -> {
        try {
            method.invoke(listener, event);
        } catch (InvocationTargetException | IllegalAccessException e) {
            throw new RuntimeException(e);
        }
    });
```